src/Componenets/Mission.js

```diff
@@ -1,23 +1,63 @@
 import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import { joinMission, leaveMission } from '../Redux/Missions/mission';
 
-const Missions = ({ missionName, missionDescription }) => (
-  <tr>
-    <td>{missionName}</td>
+const Missions = ({
+  missionId, reserved, missionName, missionDescription,
+}) => {
+  const dispatch = useDispatch();
+  const reserveMission = () => {
+    dispatch(joinMission(missionId));
+  };
+  const cancelReservation = () => {
+    dispatch(leaveMission(missionId));
+  };
 
-    <td>{missionDescription}</td>
+  return (
+    <tr>
+      <td>{missionName}</td>
 
-    <td>Active</td>
+      <td>{missionDescription}</td>
 
-    <td>
-      <button type="button" className="btn btn-secondary">
-        Join Mission
-      </button>
-    </td>
-  </tr>
-);
+      {reserved && (
+        <td>
+          <div className="badge-secondary">Not A Member</div>
+        </td>
+      )}
```

Comment:
- [ ]  **Awesome Job on The Conditional Rendering !! 💯 However, You are passing ``` reserved ``` as a prop in your ``` MissionList``` Parent Component and setting it's initial value to ``` false ```  This is a cool idea, but there will be an issue when the component re-renders. You can apply hooks such as ``` useEffect () ``` or use the global state ``` state.profile.joined ``` Property to check for reservations made on the rocket! Just like The Reservations on the Rocket Component.**